src/Language/DifferentialDatalog/Error.hs

```diff
@@ -0,0 +1,58 @@
+{-
+Copyright (c) 2020 VMware, Inc.
+SPDX-License-Identifier: MIT
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-}
+
+{-# LANGUAGE FlexibleContexts, TupleSections #-}
+
+module Language.DifferentialDatalog.Error where
+
+import Control.Monad.Except
+import Data.List
+import Language.DifferentialDatalog.Pos
+import Language.DifferentialDatalog.Name
+import Language.DifferentialDatalog.Syntax
+
+errBrief :: (MonadError String me) => Pos -> String -> me a
+errBrief p e = throwError $ spos p ++ ": " ++ e
+
+err :: (MonadError String me) => Pos -> DatalogProgram -> String -> me a
+err p d e = throwError $ sposFragment p (progSources d) e
+
+check :: (MonadError String me) => Bool -> Pos -> String -> me ()
```

Comment:
I guess the plan is to have versions of check and uniq that take DatalogProgram too?